x-pack/plugins/lens/public/xy_visualization/xy_config_panel.tsx

```diff
@@ -155,8 +188,68 @@ export function XyToolbar(props: VisualizationToolbarProps<State>) {
                 itemLayoutAlign="top"
                 hasDividers
               />
-            </EuiFormRow>
-          </EuiToolTip>
+            </EuiToolTip>
+          </EuiFormRow>
+          <EuiFormRow
+            display="columnCompressed"
+            label={i18n.translate('xpack.lens.xyChart.legendVisibilityLabel', {
+              defaultMessage: 'Legend display',
+            })}
+          >
+            <EuiButtonGroup
+              legend={i18n.translate('xpack.lens.xyChart.legendVisibilityLabel', {
+                defaultMessage: 'Legend display',
+              })}
+              name="legendDisplay"
+              buttonSize="compressed"
+              className="eui-displayInlineBlock"
+              options={legendOptions}
+              idSelected={legendOptions.find(({ value }) => value === legendMode)!.id}
+              onChange={(optionId) => {
+                const newMode = legendOptions.find(({ id }) => id === optionId)!.value;
+                if (newMode === 'auto') {
+                  props.setState({
+                    ...props.state,
+                    legend: { ...props.state.legend, isVisible: true, showSingleSeries: false },
+                  });
+                } else if (newMode === 'show') {
+                  props.setState({
+                    ...props.state,
+                    legend: { ...props.state.legend, isVisible: true, showSingleSeries: true },
+                  });
+                } else if (newMode === 'hide') {
+                  props.setState({
+                    ...props.state,
+                    legend: { ...props.state.legend, isVisible: false, showSingleSeries: false },
+                  });
+                }
+              }}
+            />
+          </EuiFormRow>
+          <EuiFormRow
+            display="columnCompressed"
+            label={i18n.translate('xpack.lens.xyChart.legendPositionLabel', {
+              defaultMessage: 'Legend position',
+            })}
+          >
+            <EuiSelect
```

Comment:
Is it actually possible to change this to a button group? Will it fit in the space?

Reply:
It doesn't fit in the default space:
<img width="418" alt="Screenshot 2020-07-21 at 11 26 08" src="https://user-images.githubusercontent.com/1508364/88037368-1264cf80-cb45-11ea-840e-3352a8e8e3fa.png">